templates/routing-legacy/main.go

```diff
@@ -35,41 +39,149 @@ func solver(
 	ctx context.Context,
 	input schema.FleetInput,
 	options options,
-) (runSchema.Output, error) {
+) (FleetOutput, error) {
 	nextrouteInput, err := input.ToNextRoute()
 	if err != nil {
 		panic(err)
 	}
 
-	solveOptions := options.Solve
 	if input.Options != nil && input.Options.Solver != nil &&
 		input.Options.Solver.Limits != nil {
-		duration, err := time.ParseDuration(input.Options.Solver.Limits.Duration)
-		if err != nil {
-			return runSchema.Output{}, err
+		if input.Options.Solver.Limits.Duration != "" {
+			duration, err := time.ParseDuration(input.Options.Solver.Limits.Duration)
+			if err != nil {
+				return FleetOutput{}, err
+			}
+			options.Solve.Duration = duration
 		}
-
-		solveOptions.Duration = duration
 	}
 
 	model, err := factory.NewModel(nextrouteInput, options.Model)
 	if err != nil {
-		return runSchema.Output{}, err
+		return FleetOutput{}, err
+	}
+
+	distanceExpression := makeDistanceExpression(input)
+	for _, v := range model.VehicleTypes() {
+		v.SetData(distanceData{
+			distance: distanceExpression,
+		})
 	}
 
 	solver, err := nextroute.NewParallelSolver(model)
 	if err != nil {
-		return runSchema.Output{}, err
+		return FleetOutput{}, err
+	}
+
+	solutionsCount := "last"
+	runSolutions, err := run.ParseSolutions(solutionsCount)
```

Comment:
Can't you just do:

`runSolutions := run.All`

Reply:
yep. did not see that. thanks